trikot-viewmodels-declarative-compiler/viewmodels-declarative-compiler-core/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compiler/VMDScanner.kt

```diff
@@ -30,10 +34,15 @@ class VMDScanner(
                     resolver = resolver
                 )
 
+                val viewModelProperties = getViewModelProperties(vmdCodeGenerator, resolver, viewModelInterface)
+                val viewModelAbstractMethods = getViewModelAbstractMethods(vmdCodeGenerator, resolver, viewModelInterface)
+
                 ViewModelMetaData(
                     viewModelInterface = viewModelInterface,
                     publishedProperty = entry.value.map { it as KSPropertyDeclaration },
-                    superClass = superClassDeclaration
+                    superClass = superClassDeclaration,
+                    allFieldsArePublished = entry.value.size == viewModelProperties.count(),
```

Comment:
What about properties such as
```
    val someString: String
        get() = "YOLO"
```

This doesn't need to be counted as it does not need to be overriden

Reply:
That would be a weird thing to have on a viewmodel interface, but I agree that this case should be covered, just in case.